src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -695,8 +704,9 @@ ov::SoPtr<ov::ICompiledModel> ov::npuw::CompiledModel::compile_submodel(const st
 
 void ov::npuw::CompiledModel::dump_on_fail(std::size_t id, const std::string& device_to_try, const char* extra) {
     const std::string dof_opt = m_cfg.get<::intel_npu::NPUW_DUMP_SUBS_ON_FAIL>();
+    const std::size_t last_idx = m_compiled_submodels.size() - 1;
```

Comment:
surely, it should be `end_idx` to avoid these recurring `-1`s.

Reply:
Fixed